### PR TITLE
Export to CSV

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: benchto-service/src/main/webapp/js/lib/ng-csv.js [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+        element.bind('click', function (e) {
+          scope.buildCSV().then(function (csv) {
+            doClick();
+          });
+          scope.$apply();
+        });
+      }
+    };
+  }]);
+})(window, document);
```
Comment (kokosing) [inline benchto-service/src/main/webapp/js/lib/ng-csv.js]: no new line :)

File: benchto-service/src/main/webapp/js/controllers.js [hunk 2]
```diff
@@ -222,7 +222,7 @@
                 $modalInstance.dismiss('cancel');
             }
         }])
-        .controller('CartCompareNavBarCtrl', ['$scope', '$location', 'CartCompareService', function ($scope, $location, CartCompareService) {
+        .controller('CartCompareNavBarCtrl', ['$scope', '$location', '$q', 'CartCompareService', 'BenchmarkService', function ($scope, $location, $q, CartCompareService, BenchmarkService) {
```
Comment (kokosing) [inline benchto-service/src/main/webapp/js/controllers.js]: what is `q`?

File: benchto-service/src/main/webapp/js/controllers.js [hunk 3]
```diff
@@ -243,6 +243,95 @@
...
+
+                $scope.benchmarkRuns = [];
+                for (var i in sequenceIds) {
+                    BenchmarkService.loadBenchmarkRun(names[i], sequenceIds[i])
+                        .then(function (benchmarkRun) {
+                            $scope.benchmarkRuns.push(benchmarkRun);
+                            prepareCsv();
+                        });
+                }
+                console.log("x " + $scope.benchmarkRuns );
```
Comment (kokosing) [inline benchto-service/src/main/webapp/js/controllers.js]: remove

File: benchto-service/src/main/webapp/js/controllers.js [hunk 4]
```diff
@@ -243,6 +243,95 @@
...
+                    if (sequenceIds.length != $scope.benchmarkRuns.length) {
+                        return; // not all benchmarkRuns are loaded yet
+                    }
+
+                    $scope.benchmarkRuns = _.sortBy($scope.benchmarkRuns, function(run) {
+                       return run.uniqueName + run.sequence_id;
+                    })
+
+                    var benchmarkRunsHelper = new BenchmarkRunsHelper($scope.benchmarkRuns, []);
+                    //var aggregatedMeasurementsKeys = benchmarkRunsHelper.aggregatedExecutionsMeasurementKeys();
```
Comment (kokosing) [inline benchto-service/src/main/webapp/js/controllers.js]: remove

File: benchto-service/src/main/webapp/js/controllers.js [hunk 5]
```diff
@@ -243,6 +243,95 @@
...
+                        .then(function (benchmarkRun) {
+                            $scope.benchmarkRuns.push(benchmarkRun);
+                            prepareCsv();
+                        });
+                }
+                console.log("x " + $scope.benchmarkRuns );
+
+                var csvDataDefer = $q.defer();
+                var prepareCsv = function () {
+                    if (sequenceIds.length != $scope.benchmarkRuns.length) {
```
Comment (kokosing) [inline benchto-service/src/main/webapp/js/controllers.js]: maybe button should get enabled after benchmarks are loaded?

File: benchto-service/src/main/webapp/js/controllers.js [hunk 6]
```diff
@@ -243,6 +243,95 @@
...
+                        return; // not all benchmarkRuns are loaded yet
+                    }
+
+                    $scope.benchmarkRuns = _.sortBy($scope.benchmarkRuns, function(run) {
+                       return run.uniqueName + run.sequence_id;
+                    })
+
+                    var benchmarkRunsHelper = new BenchmarkRunsHelper($scope.benchmarkRuns, []);
+                    //var aggregatedMeasurementsKeys = benchmarkRunsHelper.aggregatedExecutionsMeasurementKeys();
+                    var aggregatedMeasurementsKeys = ['duration']; // let's stick to duration for now
```
Comment (kokosing) [inline benchto-service/src/main/webapp/js/controllers.js]: I would prefer to have all the measurements, it is easy to filter afterwards

File: benchto-service/src/main/webapp/js/lib/ng-csv.js (hunk 1 above)
Comment (losipiuk) [inline benchto-service/src/main/webapp/js/lib/ng-csv.js]: this is lib

File: benchto-service/src/main/webapp/js/controllers.js (hunk 5 above)
Comment (losipiuk) [inline benchto-service/src/main/webapp/js/controllers.js]: gfy

File: benchto-service/src/main/webapp/js/controllers.js (hunk 6 above)
Comment (losipiuk) [inline benchto-service/src/main/webapp/js/controllers.js]: I took just duration as @pnowojski states that other measurements are useless and do not correspond to reality.

File: benchto-service/src/main/webapp/js/controllers.js (hunk 4 above)
Comment (losipiuk) [inline benchto-service/src/main/webapp/js/controllers.js]: done

**Comment by ArturGajowy** [comment]
We merge this? Or drop?

**Comment by losipiuk** [comment]
It is merged

**Comment by losipiuk** [comment]
Afaik

**Comment by ArturGajowy** [comment]
Github thinks otherwise :) Please check and close

**Pull request closed without merging**